Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM python:3.8
+
+# Copy current sources
+COPY . /src
+
+# install bets
+RUN pip install /src
+
+# install all possibly dependencies for plugins
+RUN pip install pigar
+RUN pigar -p /src/plugins-requirements.txt -P /src
+RUN pip install -r /src/plugins-requirements.txt
```

Comment:
I don't entirely understand the need for `pigar` in this context. Would it make sense to just `pip install` beets with several extras to get all those optional dependencies?

Reply:
`pigar` scans everything inside `/src` and put all imports. Why? Because to enable some plugins user also needs to `pip install XXX` that is explained inside docs.

I'm a bit lazy to make it by hand and I've used `pigar` for it.

Reply:
That's the thing—all those dependencies are already listed as "extras" in our setup.py. So we can just use those there:
https://github.com/beetbox/beets/blob/04ea754d00e2873ae9aa2d9e07c5cefd790eaee2/setup.py#L140

Reply:
I've missed it :(

Reply:
Yes, this makes things easy. Thanks!

Reply:
@sampsyo is it possibly to install all plugins by some magic command?

Reply:
I don't know for sure, but I think the answer is no: https://github.com/beetbox/beets/blob/04ea754d00e2873ae9aa2d9e07c5cefd790eaee2/setup.py#L140

Reply:
@sampsyo I'm confused. Why? Because I will introduce one more list of plugins.

:(

Reply:
Yeah, that's not great. But TBH I prefer it a bit to attempting to automatically re-discover dependencies that we have already pinned down based on import names.

Reply:
@sampsyo defentply. Maybe introducing a new extra "all plugins" is the best way?

Reply:
No objection there. It seems like there needs to be an "all" list either in the Dockerfile or in setup.py, and I don't immediately see a reason to prioritize one or the other.

Reply:
@sampsyo any objections why it can't install all dependencies? :)

Honestly docs a bit confused because I read that I need run `pip install requests` and this sort of things, and I feel that plugins aren't so deep integrated.

Now I see that plugins is very deep integrated, and for me it isn't clear why have you used extras here.

Reply:
Well, that is what pip extras are meant to express: things you might need if you use certain features, but you're not guaranteed to need. Making them optional dependencies means that you don't need to install as many libraries to get started.